controller/src/main/java/io/pravega/controller/store/stream/PersistentStreamBase.java

```diff
@@ -249,14 +249,22 @@ private boolean greaterThan(Map<Long, Long> cut1, Map<StreamSegmentRecord, Integ
     }
 
     @Override
-    public CompletableFuture<Boolean> isStreamCutStrictlyGreaterThan(Map<Long, Long> streamcut1, Map<Long, Long> streamcut2) {
+    public CompletableFuture<Integer> compareStreamCuts(Map<Long, Long> streamcut1, Map<Long, Long> streamcut2) {
         CompletableFuture<ImmutableMap<StreamSegmentRecord, Integer>> span1Future = computeStreamCutSpan(streamcut1);
         CompletableFuture<ImmutableMap<StreamSegmentRecord, Integer>> span2Future = computeStreamCutSpan(streamcut2);
         return CompletableFuture.allOf(span1Future, span2Future)
                     .thenApply(v -> {
                         ImmutableMap<StreamSegmentRecord, Integer> span1 = span1Future.join();
                         ImmutableMap<StreamSegmentRecord, Integer> span2 = span2Future.join();
-                        return greaterThan(streamcut1, span1, streamcut2, span2);
+                        boolean gt = greaterThan(streamcut1, span1, streamcut2, span2);
```

Comment:
Instead of 2 method calls to `greaterThan` here with reverse parameters, we could perhaps have another method like compareCuts() that makes the same 2 calls as here and returns the appropriate enum.

Reply:
the compare method is the one thats calling these two methods back to back and making use of their result.
this is not a costly method, its a simple in memory comparison of two streamcuts.

we could create a common method which will need to do the same comparisons that these two methods do and there wouldnt be much room for optimizing that.

Reply:
Yes a common method would be better though internally it would make the same calls as is being done here, it keeps the code more readable. It may be good to add a javadoc comment to this method and that new method explaining what each one achieves.

Reply:
so a common method is just a wrapper over these two methods, right?.. isnt this method exactly that in a way?